app/src/main/java/ciserver/Pipeline.java

```diff
@@ -8,65 +11,64 @@ enum PipelineStatus {
 	InProgress
 }
 
+interface PipelineComponent {
+	public PipelineStatus execute(String pipelineDir, PushEvent event);
+}
+
 /**
  * A PipelineInstance is responsible for the whole CI-pipeline
  * for a specific commit.
  */
 class Pipeline {
-	String pipelineDir;
-	Commit commit;
+	final String pipelineDir;
+	final PushEvent event;
 	PipelineStatus status;
```

Comment:
Fields should be private imo.

```suggestion

	private final String pipelineDir;
	private final PushEvent event;
```

Reply:
I thought they were automatically, good catch!